File: aicontentfactory/backend/app/api/v1/routes/ai.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.infrastructure.database import get_db
from app.domain.ai import schemas, services

router = APIRouter()


@router.post("/analyze", response_model=schemas.AnalysisResponse)
async def analyze_signal(payload: schemas.AnalysisRequest, db: Session = Depends(get_db)):
    result = services.analyze_signal(db, payload)
    return {"data": result}


@router.post("/generate")
async def generate_article(payload: schemas.GenerateRequest):
    result = services.generate_article(payload)
    return {"data": result}


@router.post("/polish")
async def polish_article(payload: schemas.PolishRequest):
    result = services.polish_article(payload)
    return {"data": result}


@router.post("/detect-ai")
async def detect_ai(payload: schemas.DetectAIRequest):
    result = services.detect_ai(payload)
    return {"data": result}


@router.get("/providers")
async def list_providers():
    providers = services.list_providers()
    return {"data": providers}


@router.post("/providers/{provider_id}/test")
async def test_provider(provider_id: str):
    result = services.test_provider(provider_id)
    return {"data": result}

File: aicontentfactory/frontend/src/stores/ai.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { aiApi } from '@/api'

export const useAIStore = defineStore('ai', () => {
  const providers = ref<unknown[]>([])
  const loading = ref(false)

  const analyze = async (data: unknown) => {
    try {
      const response = await aiApi.analyze(data)
      return response.data.data
    } catch (error) {
      console.error('Failed to analyze:', error)
      throw error
    }
  }

  const generate = async (data: unknown) => {
    try {
      const response = await aiApi.generate(data)
      return response.data.data
    } catch (error) {
      console.error('Failed to generate:', error)
      throw error
    }
  }

  const polish = async (data: unknown) => {
    try {
      const response = await aiApi.polish(data)
      return response.data.data
    } catch (error) {
      console.error('Failed to polish:', error)
      throw error
    }
  }

  const detectAI = async (data: unknown) => {
    try {
      const response = await aiApi.detectAI(data)
      return response.data.data
    } catch (error) {
      console.error('Failed to detect AI:', error)
      throw error
    }
  }

  const listProviders = async () => {
    loading.value = true
    try {
      const response = await aiApi.listProviders()
      providers.value = response.data.data
    } catch (error) {
      console.error('Failed to list providers:', error)
    } finally {
      loading.value = false
    }
  }

  const testProvider = async (id: string) => {
    try {
      const response = await aiApi.testProvider(id)
      return response.data.data
    } catch (error) {
      console.error('Failed to test provider:', error)
      throw error
    }
  }

  return {
    providers,
    loading,
    analyze,
    generate,
    polish,
    detectAI,
    listProviders,
    testProvider,
  }
})
